tests/prophet/test_prophet_model_export.py

```diff
@@ -0,0 +1,455 @@
+
+
+def test_model_log_without_specified_conda_env_uses_default_env_with_expected_dependencies(
+    prophet_model,
+):
+    artifact_path = "model"
+    with mlflow.start_run():
+        mlflow.prophet.log_model(prophet_model.model, artifact_path)
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+    _assert_pip_requirements(model_uri, mlflow.prophet.get_default_pip_requirements())
+
+
+@pytest.mark.large
+def test_pyfunc_serve_and_score(prophet_model):
+
+    artifact_path = "model"
+    with mlflow.start_run():
+        mlflow.prophet.log_model(prophet_model.model, artifact_path)
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+    local_predict = prophet_model.model.predict(
+        prophet_model.model.make_future_dataframe(FORECAST_HORIZON)
+    )
+
+    # cast to string representation of datetime series, otherwise will default cast to Unix time
+    # which Prophet does not support for encoding
+    inference_data = (
+        prophet_model.model.make_future_dataframe(FORECAST_HORIZON)["ds"]
+        .dt.strftime(INFER_FORMAT)
+        .to_frame(name="ds")
+    )
+
+    resp = pyfunc_serve_and_score_model(
+        model_uri,
+        data=inference_data,
+        content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON_RECORDS_ORIENTED,
+    )
+
+    scores = pd.read_json(resp.content, orient="records")
+
+    # predictions are deterministic, but yhat_lower, yhat_upper are non-deterministic based on
+    # stan build underlying environment. Seed value only works for reproducibility of yhat.
+    # see: https://github.com/facebook/prophet/issues/1124
+    pd.testing.assert_series_equal(
+        left=local_predict["yhat"], right=scores["yhat"], check_dtype=True
+    )
+
+
+@pytest.mark.large
+def test_metric_and_parameter_logging(prophet_model, model_path):
+
+    from prophet import serialize
+    from prophet.diagnostics import cross_validation, performance_metrics
+    from mlflow.tracking import MlflowClient
+
+    def extract_params(model):
+        return {attr: getattr(model, attr) for attr in serialize.SIMPLE_ATTRIBUTES}
+
+    with mlflow.start_run():
+        params = extract_params(prophet_model.model)
+        run = mlflow.active_run().info.run_id
+        metric_values = ["mse", "rmse", "mae", "mape", "mdape", "smape", "coverage"]
+        metrics_raw = cross_validation(
+            prophet_model.model, "365 days", "180 days", "710 days", "threads", disable_tqdm=True
+        )
+        cv_metrics = performance_metrics(metrics_raw)
+
+        metrics = {}
+        for metric in metric_values:
+            metrics[metric] = cv_metrics[metric].mean()
+
+        mlflow.prophet.save_model(prophet_model.model, model_path)
+        mlflow.log_params(params)
+        mlflow.log_metrics(metrics)
+
+    client = MlflowClient()
+    run_id = client.get_run(run).info.run_id
+    logged_metrics = client.get_metric_history(run_id, "rmse")
+    assert logged_metrics[0].value == metrics["rmse"]
```

Comment:
We probably don't need this test case since it isn't testing functionality introduced in the `mlflow.prophet` model flavor. This may be more applicable for autologging in the future.

Reply:
Removed